samtranslator/feature_toggle/feature_toggle.py

```diff
@@ -18,50 +18,53 @@ class FeatureToggle:
             LOG.warning("Feature '{}' not available in Feature Toggle Config.".format(feature_name))
             return False
-        stage_config = self.feature_config.get(feature_name, {}).get(stage, {})
-        if not stage_config:
-            LOG.info("Stage '{}' not enabled for Feature '{}'.".format(stage, feature_name))
-            return False
-        region_config = stage_config.get(region, {}) if region in stage_config else stage_config.get("default", {})
-        is_enabled = region_config.get("enabled", False)
-        LOG.info("Feature '{}' is enabled: '{}'".format(feature_name, is_enabled))
-        return is_enabled
 
-    def is_enabled_for_account_in_region(self, feature_name, stage, account_id, region="default"):
-        """
-        To check if feature is available for a particular account or not.
-        :param feature_name: name of feature
-        :param stage: stage where SAM is running
-        :param account_id: account_id who is executing SAM template
-        :param region: region in which SAM is running
-        :return:
-        """
-        if feature_name not in self.feature_config:
-            LOG.warning("Feature '{}' not available in Feature Toggle Config.".format(feature_name))
+        stage = self.stage
+        region = self.region
+        account_id = self.account_id
+        if not stage or not region or not account_id:
+            LOG.warning(
+                "One or more of stage, region and account_id is not set. Feature '{}' not enabled.".format(feature_name)
+            )
             return False
+
         stage_config = self.feature_config.get(feature_name, {}).get(stage, {})
         if not stage_config:
             LOG.info("Stage '{}' not enabled for Feature '{}'.".format(stage, feature_name))
             return False
-        account_config = stage_config.get(account_id) if account_id in stage_config else stage_config.get("default", {})
-        region_config = (
-            account_config.get(region, {}) if region in account_config else account_config.get("default", {})
-        )
-        is_enabled = region_config.get("enabled", False)
+
+        if account_id in stage_config:
+            account_config = stage_config[account_id]
+            region_config = account_config[region] if region in account_config else account_config.get("default", {})
+        else:
+            region_config = stage_config[region] if region in stage_config else stage_config.get("default", {})
+
+        if "enabled-%" in region_config:
+            # Percentage-based enablement
+            # Assumption: account_id is uniformly distributed
+            # account_id is calculated into one of 100 partitions (0-99)
+            # if partition < enabled_percent, we consider the feature is enabled for this given account_id
+            enabled_percent = region_config["enabled-%"]
+            partition = int(account_id) % 100
+            is_enabled = partition < enabled_percent
```

Comment:
Can we automate this dial up or enablement by specifiying start and end time in UTC. this way its automatically controlled dialup once the config is on appconfig, without needing any pushes for dialup.

Reply:
If we have plans for dial-up, we _could_ make this generic from the get-go. The percentage is linear between start/end percentages during the time interval, and an always-on dial is an open interval (e.g. end is infinite).

Reply:
Let's just handle the simple percentage-based enablement in this PR first. We can consider automated dial up later